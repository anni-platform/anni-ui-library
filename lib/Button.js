"use strict";function _interopDefault(n){return n&&"object"==typeof n&&"default"in n?n.default:n}Object.defineProperty(exports,"__esModule",{value:!0});var styled=require("styled-components"),styled__default=_interopDefault(styled);function _taggedTemplateLiteral(n,t){return t||(t=n.slice(0)),Object.freeze(Object.defineProperties(n,{raw:{value:Object.freeze(t)}}))}var EASE_OUT_EXPO="cubic-bezier(0.19, 1, 0.22, 1)",breakpointSizes={xl:1600,lg:1200,md:960,sm:570};function _templateObject3(){var n=_taggedTemplateLiteral(["\n    @media (min-width: ","em) {\n      ",";\n    }\n  "]);return _templateObject3=function(){return n},n}function _templateObject2(){var n=_taggedTemplateLiteral(["\n    padding-top: ",";\n    padding-right: ",";\n    padding-bottom: ",";\n    padding-left: ",";\n  "]);return _templateObject2=function(){return n},n}function _templateObject(){var n=_taggedTemplateLiteral(["\n    margin-top: ",";\n    margin-right: ",";\n    margin-bottom: ",";\n    margin-left: ",";\n  "]);return _templateObject=function(){return n},n}var utils={margin:function(){return styled.css(_templateObject(),function(n){var t=n.mt;return t?"".concat(t,"px"):"0"},function(n){var t=n.mr;return t?"".concat(t,"px"):"0"},function(n){var t=n.mb;return t?"".concat(t,"px"):"0"},function(n){var t=n.ml;return t?"".concat(t,"px"):"0"})},padding:function(){return styled.css(_templateObject2(),function(n){var t=n.pt;return t?"".concat(t,"px"):"0"},function(n){var t=n.pr;return t?"".concat(t,"px"):"0"},function(n){var t=n.pb;return t?"".concat(t,"px"):"0"},function(n){var t=n.pl;return t?"".concat(t,"px"):"0"})}},Above=Object.keys(breakpointSizes).reduce(function(n,t){return n[t]=function(){return styled.css(_templateObject3(),breakpointSizes[t]/16,styled.css.apply(void 0,arguments))},n},{});function _templateObject$1(){var n=_taggedTemplateLiteral(["\n  align-items: center;\n  backface-visibility: hidden;\n  background: ",";\n  border: ",";\n  border: ",";\n  border-radius: 80px;\n  color: ",";\n  cursor: pointer;\n  display: inline-flex;\n  flex-direction: ",";\n  font-family: 'Apercu Bold', sans-serif;\n  font-size: ",";\n  justify-content: center;\n  letter-spacing: 1px;\n  line-height: 1;\n  outline: none;\n  padding: ",";\n  text-decoration: none;\n  text-transform: uppercase;\n  transition: 400ms ",";\n  z-index: 2;\n  ",";\n\n  &:active {\n    transform: scale(0.998);\n  }\n\n  &:focus,\n  &:hover {\n    transform: scale(1.02);\n\n    &:disabled,\n    &[disabled] {\n      cursor: default;\n      transform: scale(1);\n    }\n  }\n\n  svg {\n    ",";\n    fill: ",";\n    height: ",";\n    margin-right: ",";\n    margin-bottom: ",";\n    stroke: ",";\n    stroke: ",";\n    stroke-width: ",";\n    stroke-width: ",";\n    transition: 400ms ",";\n    width: ",";\n  }\n"]);return _templateObject$1=function(){return n},n}var Button=styled__default.button(_templateObject$1(),function(n){return!n.primary||n.link?"none":"".concat(n.theme.colorPrimary)},function(n){return n.noBorder||n.link?"none":"2px solid ".concat(n.theme.colorPrimary)},function(n){return n.stacked&&"none"},function(n){return n.theme.textPrimary},function(n){return n.stacked&&"column"},function(n){return n.stacked?"12px":"13px"},function(n){return n.full?"0":"12px 24px"},EASE_OUT_EXPO,utils.margin,utils.padding,function(n){return n.fill?n.theme.textPrimary:"transparent"},function(n){return n.iconSize?"".concat(n.iconSize,"px"):"18px"},function(n){return n.stacked?"0":"6px"},function(n){return n.stacked?"6px":"0"},function(n){return n.fill?"transparent":"".concat(n.theme.textPrimary)},function(n){return n.stacked&&"".concat(n.theme.textPrimary)},function(n){return n.strokeWidth?"".concat(n.strokeWidth,"px"):"6px"},function(n){return n.fill&&"6px"},EASE_OUT_EXPO,function(n){return n.iconSize?"".concat(n.iconSize,"px"):"18px"});exports.Button=Button;
