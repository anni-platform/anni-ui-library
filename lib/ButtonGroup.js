"use strict";function _interopDefault(t){return t&&"object"==typeof t&&"default"in t?t.default:t}Object.defineProperty(exports,"__esModule",{value:!0});var styled=require("styled-components"),styled__default=_interopDefault(styled);function _taggedTemplateLiteral(t,e){return e||(e=t.slice(0)),Object.freeze(Object.defineProperties(t,{raw:{value:Object.freeze(e)}}))}var breakpointSizes={xl:1600,lg:1200,md:960,sm:570};function _templateObject3(){var t=_taggedTemplateLiteral(["\n    @media (min-width: ","em) {\n      ",";\n    }\n  "]);return _templateObject3=function(){return t},t}function _templateObject2(){var t=_taggedTemplateLiteral(["\n    padding-top: ",";\n    padding-right: ",";\n    padding-bottom: ",";\n    padding-left: ",";\n  "]);return _templateObject2=function(){return t},t}function _templateObject(){var t=_taggedTemplateLiteral(["\n    margin-top: ",";\n    margin-right: ",";\n    margin-bottom: ",";\n    margin-left: ",";\n  "]);return _templateObject=function(){return t},t}var utils={margin:function(){return styled.css(_templateObject(),function(t){var e=t.mt;return e?"".concat(e,"px"):"0"},function(t){var e=t.mr;return e?"".concat(e,"px"):"0"},function(t){var e=t.mb;return e?"".concat(e,"px"):"0"},function(t){var e=t.ml;return e?"".concat(e,"px"):"0"})},padding:function(){return styled.css(_templateObject2(),function(t){var e=t.pt;return e?"".concat(e,"px"):"0"},function(t){var e=t.pr;return e?"".concat(e,"px"):"0"},function(t){var e=t.pb;return e?"".concat(e,"px"):"0"},function(t){var e=t.pl;return e?"".concat(e,"px"):"0"})}},Above=Object.keys(breakpointSizes).reduce(function(t,e){return t[e]=function(){return styled.css(_templateObject3(),breakpointSizes[e]/16,styled.css.apply(void 0,arguments))},t},{});function _templateObject$1(){var t=_taggedTemplateLiteral(["\n  ",";\n  display: flex;\n  justify-content: ",";\n  button {\n    margin: 0 8px 8px 0;\n  }\n"]);return _templateObject$1=function(){return t},t}var ButtonGroup=styled__default.div(_templateObject$1(),utils.margin,function(t){return t.center?"center":"flex-start"});exports.ButtonGroup=ButtonGroup;
