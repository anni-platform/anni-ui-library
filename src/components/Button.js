import styled from 'styled-components';
import utils from '../utils';
import { CHARCOAL, COPPER, EASE_OUT_EXPO } from '../constants';

export const Button = styled.button`
  align-items: center;
  backface-visibility: hidden;
  background: ${props => (!props.primary || props.link ? 'none' : `${COPPER}`)};
  border: ${props =>
    props.noBorder || props.link ? 'none' : `2px solid ${COPPER}`};
  border: ${props => props.stacked && 'none'};
  border-radius: 80px;
  color: ${CHARCOAL};
  cursor: pointer;
  display: inline-flex;
  flex-direction: ${props => props.stacked && 'column'};
  font-family: 'Apercu Bold', sans-serif;
  font-size: ${props => (props.stacked ? '12px' : '13px')};
  justify-content: center;
  letter-spacing: 1px;
  line-height: 1;
  outline: none;
  padding: ${props => (props.full ? '0' : '12px 24px')};
  text-decoration: none;
  text-transform: uppercase;
  transition: 400ms ${EASE_OUT_EXPO};
  z-index: 2;
  ${utils.margin};

  &:active {
    transform: scale(0.998);
  }

  &:focus,
  &:hover {
    transform: scale(1.02);

    &:disabled,
    &[disabled] {
      cursor: default;
      transform: scale(1);
    }
  }

  svg {
    ${utils.padding};
    fill: ${props => (props.fill ? `${CHARCOAL}` : 'transparent')};
    height: ${props => (props.iconSize ? `${props.iconSize}px` : '18px')};
    margin-right: ${props => (props.stacked ? '0' : '6px')};
    margin-bottom: ${props => (props.stacked ? '6px' : '0')};
    stroke: ${props => (props.fill ? 'transparent' : `${CHARCOAL}`)};
    stroke: ${props => props.stacked && `${CHARCOAL}`};
    stroke-width: ${props =>
      props.strokeWidth ? `${props.strokeWidth}px` : '6px'};
    stroke-width: ${props => props.fill && '6px'};
    transition: 400ms ${EASE_OUT_EXPO};
    width: ${props => (props.iconSize ? `${props.iconSize}px` : '18px')};
  }
`;
