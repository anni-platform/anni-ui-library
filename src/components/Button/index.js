import styled from 'styled-components';
import utils from 'shared/utils';
import { EASE_OUT_EXPO } from 'shared/constants';

export const Button = styled.button`
  align-items: center;
  backface-visibility: hidden;
  background: ${props =>
    !props.primary || props.link ? 'none' : `${props.theme.colorPrimary}`};
  border: ${props =>
    props.noBorder || props.link
      ? 'none'
      : `2px solid ${props.theme.colorPrimary}`};
  border: ${props => props.stacked && 'none'};
  border-radius: 80px;
  color: ${props => props.theme.textPrimary};
  cursor: pointer;
  display: inline-flex;
  flex-direction: ${props => props.stacked && 'column'};
  font-family: 'Apercu Bold', sans-serif;
  font-size: ${props => (props.stacked ? '12px' : '13px')};
  justify-content: center;
  letter-spacing: 1px;
  line-height: 1;
  outline: none;
  padding: ${props => (props.full ? '0' : '12px 24px')};
  text-decoration: none;
  text-transform: uppercase;
  transition: 400ms ${EASE_OUT_EXPO};
  z-index: 2;
  ${utils.margin};

  &:active {
    transform: scale(0.998);
  }

  &:focus,
  &:hover {
    transform: scale(1.02);

    &:disabled,
    &[disabled] {
      cursor: default;
      transform: scale(1);
    }
  }
`;

export const NakedButton = styled(Button)`
  background: none;
  border: none;
  display: flex;
  flex-direction: column;

  span {
    margin: 4px 0 0 0;
  }
`;
