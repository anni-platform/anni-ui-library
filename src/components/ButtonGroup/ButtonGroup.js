import styled from 'styled-components';
import utils from '../../utils';

export const ButtonGroup = styled.div`
  ${utils.margin};
  display: flex;
  justify-content: ${props => (props.center ? 'center' : 'flex-start')};
  button {
    margin: 0 8px 8px 0;
  }
`;
