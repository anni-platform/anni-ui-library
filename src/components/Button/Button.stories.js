import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { Button, NakedButton } from './Button';
import { OutlineIcon } from '../Icon/Icon';

storiesOf('Form/Buttons', module)
  .add('default', () => (
    <Button onClick={action('clicked')}>Default Button</Button>
  ))
  .add('primary', () => (
    <Button onClick={action('clicked')} primary>
      Primary Button
    </Button>
  ))
  .add('naked', () => (
    <NakedButton onClick={action('clicked')}>
      <OutlineIcon name="logo" />
    </NakedButton>
  ));
