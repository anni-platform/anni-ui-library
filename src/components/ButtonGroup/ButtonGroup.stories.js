import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { Button } from '../Button/Button';
import { ButtonGroup } from './ButtonGroup';

storiesOf('Controls/ButtonGroup', module)
  .add('default', () => (
    <ButtonGroup>
      <Button onClick={action('clicked')}>Button A</Button>
      <Button onClick={action('clicked')}>Button B</Button>
    </ButtonGroup>
  ))
  .add('center', () => (
    <ButtonGroup center>
      <Button onClick={action('clicked')}>Button A</Button>
      <Button onClick={action('clicked')}>Button B</Button>
    </ButtonGroup>
  ));
