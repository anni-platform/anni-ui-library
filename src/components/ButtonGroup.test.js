import React from 'react';
import renderer from 'react-test-renderer';
import { ButtonGroup } from './ButtonGroup';
import { Button } from './Button';
import defaultTheme from '../themes/default';

test('ButtonGroup default', () => {
  expect(
    renderer
      .create(<ButtonGroup theme={defaultTheme}>Perform action</ButtonGroup>)
      .toJSON()
  ).toMatchSnapshot();
});

test('ButtonGroup center', () => {
  expect(
    renderer
      .create(
        <ButtonGroup theme={defaultTheme} center>
          <Button>Perform action</Button>
        </ButtonGroup>
      )
      .toJSON()
  ).toMatchSnapshot();
});
