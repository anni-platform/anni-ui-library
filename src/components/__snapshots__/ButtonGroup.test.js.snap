// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup center 1`] = `
.c0 {
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 button {
  margin: 0 8px 8px 0;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  background: none;
  border: 2px solid undefined;
  border-radius: 80px;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Apercu Bold',sans-serif;
  font-size: 13px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1;
  outline: none;
  padding: 12px 24px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: 400ms cubic-bezier(0.19,1,0.22,1);
  transition: 400ms cubic-bezier(0.19,1,0.22,1);
  z-index: 2;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
}

.c1:active {
  -webkit-transform: scale(0.998);
  -ms-transform: scale(0.998);
  transform: scale(0.998);
}

.c1:focus,
.c1:hover {
  -webkit-transform: scale(1.02);
  -ms-transform: scale(1.02);
  transform: scale(1.02);
}

.c1:focus:disabled,
.c1:hover:disabled,
.c1:focus[disabled],
.c1:hover[disabled] {
  cursor: default;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c1 svg {
  padding-top: 0;
  padding-right: 0;
  padding-bottom: 0;
  padding-left: 0;
  fill: transparent;
  height: 18px;
  margin-right: 6px;
  margin-bottom: 0;
  stroke: undefined;
  stroke-width: 6px;
  -webkit-transition: 400ms cubic-bezier(0.19,1,0.22,1);
  transition: 400ms cubic-bezier(0.19,1,0.22,1);
  width: 18px;
}

<div
  className="c0"
>
  <button
    className="c1"
  >
    Perform action
  </button>
</div>
`;

exports[`ButtonGroup default 1`] = `
.c0 {
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 button {
  margin: 0 8px 8px 0;
}

<div
  className="c0"
>
  Perform action
</div>
`;
